src/test/plugins/shared_dictionaries_plugin_test.cpp

```diff
@@ -0,0 +1,265 @@
+    if (!sm.has_table("customer")) {
+      const auto table = load_table(_get_table_path("customer"), _chunk_size);
+      _encode_table(table);
+      sm.add_table("customer", table);
+    }
+  }
+
+  void _validate_customer_table() {
+    auto& sm = Hyrise::get().storage_manager;
+    EXPECT_TABLE_EQ_ORDERED(sm.get_table("customer"), load_table(_get_table_path("customer"), _chunk_size));
+  }
+
+  void _encode_table(std::shared_ptr<Table> table) {
+    auto chunk_encoding_spec = ChunkEncodingSpec{};
+    for (const auto& column_definition : table->column_definitions()) {
+      if (encoding_supports_data_type(EncodingType::Dictionary, column_definition.data_type)) {
+        chunk_encoding_spec.emplace_back(EncodingType::Dictionary);
+      } else {
+        chunk_encoding_spec.emplace_back(EncodingType::Unencoded);
+      }
+    }
+
+    ChunkEncoder::encode_all_chunks(table, chunk_encoding_spec);
+  }
+
+  std::shared_ptr<Table> _get_customer_table() {
+    auto& sm = Hyrise::get().storage_manager;
+    return sm.get_table("customer");
+  }
+
+  const SegmentChunkPair<pmr_string> _get_segment_chunk_pair() {
+    const auto chunk = _get_customer_table()->get_chunk(ChunkID{0});
+    const auto segment = chunk->get_segment(_market_column_id);
+    const auto dictionary_segment = std::dynamic_pointer_cast<DictionarySegment<pmr_string>>(segment);
+    return std::make_pair(dictionary_segment, chunk);
+  }
+
+  std::string _get_table_path(const std::string& table_name) { return _tables_path + table_name + _table_extension; }
+
+  static double _calc_jaccard_index(size_t union_size, size_t intersection_size) {
+    return SharedDictionariesColumnProcessor<int32_t>::_calc_jaccard_index(union_size, intersection_size);
+  }
+
+  static void _initialize_merge_plans(
+      SharedDictionariesColumnProcessor<pmr_string>& column_processor,
+      std::vector<std::shared_ptr<SharedDictionariesColumnProcessor<pmr_string>::MergePlan>>& merge_plans) {
+    column_processor._initialize_merge_plans(merge_plans);
+  }
+
+  static bool _should_merge(SharedDictionariesColumnProcessor<pmr_string>& column_processor, const double jaccard_index,
+                            const size_t current_dictionary_size, const size_t shared_dictionary_size,
+                            const std::vector<SegmentChunkPair<pmr_string>>& shared_segment_chunk_pairs) {
+    return column_processor._should_merge(jaccard_index, current_dictionary_size, shared_dictionary_size,
+                                          shared_segment_chunk_pairs);
+  }
+
+  static void _add_segment_chunk_pair(SharedDictionariesColumnProcessor<pmr_string>::MergePlan& merge_plan,
+                                      const SegmentChunkPair<pmr_string>& segment_chunk_pair, bool is_already_merged) {
+    SharedDictionariesColumnProcessor<pmr_string>::_add_segment_chunk_pair(merge_plan, segment_chunk_pair,
+                                                                           is_already_merged);
+  }
```

Comment:
What's the reasoning behind creating those four functions? 
Instead of directly calling `column_processor._initialize_merge_plans(merge_plans)` in the test, you now call `_initialize_merge_plans(column_processor, merge_plans)`
Does this give you any benefits?

Reply:
They are private functions and in order to test them, we have to use a `friend class`.
But the friend class does not work inside the TEST_F test fixtures. That's why we have the methods inside the test class.